lib/elixir/lib/module.ex

```diff
@@ -919,7 +922,7 @@ defmodule Module do
   @spec defines?(module, definition) :: boolean
   def defines?(module, {name, arity} = tuple)
       when is_atom(module) and is_atom(name) and is_integer(arity) and arity >= 0 and arity <= 255 do
-    assert_not_compiled!(:defines?, module)
+    assert_not_compiled!(__ENV__.function, module, @extra_error_msg_defines?)
```

Comment:
It seems this module attribute is only used once, so I would just put the message inside the function:

```elixir
extra_message = 
  "Use Kernel.function_exported?/3 and Kernel.macro_exported?/3 " <>
    "to check for public functions and macros instead"

assert_not_compiled!(__ENV__.function, module, extra_message)
```

Reply:
Actually, It is used twice. I misplaced the `@extra_error_msg_definitions_in` above `defines?/2`, so it may have given you the impression it was only used once.
It has been rebased, so you cannot see this in the commit history.

Reply:
there was some issue with my commit history, please have a look again